mcp_server/prowler_mcp_server/prowler_app/tools/providers.py

```diff
@@ -0,0 +1,551 @@
+"""Cloud Provider Management tools for Prowler App MCP Server.
+
+This module provides tools for managing cloud provider connections,
+including searching, connecting, and deleting providers.
+"""
+
+from typing import Any
+
+from prowler_mcp_server.prowler_app.models.providers import (
+    ProviderConnectionStatus,
+    ProvidersListResponse,
+)
+from prowler_mcp_server.prowler_app.tools.base import BaseTool
+from pydantic import Field
+
+
+class ProvidersTools(BaseTool):
+    """Tools for cloud provider management operations
+
+    Provides tools for:
+    - prowler_app_search_cloud_providers: Search and view configured cloud providers with their connection status
+    - prowler_app_connect_provider: Connect or register a cloud provider account for security scanning in Prowler
+    - prowler_app_delete_provider: Permanently remove a cloud provider from Prowler
+    """
+
+    async def search_cloud_providers(
+        self,
+        provider_id: list[str] = Field(
+            default=[],
+            description="Filter by Prowler's internal UUID(s) (v4) for the provider(s), generated when the provider is registered in the system.",
+        ),
+        provider_uid: list[str] = Field(
+            default=[],
+            description="Filter by provider's unique identifier(s), this ID is the one provided by the provider itself. Format varies by provider type: AWS Account ID (12 digits), Azure Subscription ID (UUID), GCP Project ID (string), Kubernetes namespace, GitHub username/organization, M365 domain ID, etc. All supported provider types are listed in the Prowler Hub/Prowler Documentation that you can also find in form of tools in this MCP Server",
+        ),
+        provider_type: list[str] = Field(
+            default=[],
+            description="Filter by cloud provider type. Valid values include: 'aws', 'azure', 'gcp', 'kubernetes'... For more valid values, please refer to Prowler Hub/Prowler Documentation that you can also find in form of tools in this MCP Server.",
+        ),
+        alias: str | None = Field(
+            default=None,
+            description="Search by provider alias/friendly name. Partial match supported (case-insensitive). Use this to find providers by their human-readable name (e.g., 'Production', 'Dev', 'AWS Main')",
+        ),
+        connected: (
```

Comment:
This doesnt seem to work correctly. 

API returns `connected: null` when listing providers

<img width="364" height="169" alt="Image" src="https://github.com/user-attachments/assets/c9c3ca1b-9756-46bd-b31b-b6dd82c23434" />

But from the tool, setting `connected: false` for `prowler_app_search_cloud_providers` returns empty results.

<img width="933" height="508" alt="Image" src="https://github.com/user-attachments/assets/88f1eb83-0954-4af1-bcd1-b3a0a9a15f38" />